pkg/autoscaler/autoscaler/util.go

```diff
@@ -146,3 +147,51 @@ func genMetricsEndpoint(tac *v1alpha1.TidbClusterAutoScaler) (string, error) {
 	}
 	return fmt.Sprintf("http://%s-prometheus.%s.svc:9090", tac.Spec.Monitor.Name, tac.Spec.Monitor.Namespace), nil
 }
+
+func autoscalerToStrategy(tac *v1alpha1.TidbClusterAutoScaler) *pdapi.Strategy {
+	strategy := &pdapi.Strategy{
+		Resources: make([]*pdapi.Resource, 0, len(tac.Spec.Resources)),
+	}
+
+	for _, res := range tac.Spec.Resources {
+		count := uint64(*res.Count)
+		strategy.Resources = append(strategy.Resources, &pdapi.Resource{
+			CPU:          res.CPU.AsDec().UnscaledBig().Uint64(),
+			Memory:       res.Memory.AsDec().UnscaledBig().Uint64(),
+			Storage:      res.Storage.AsDec().UnscaledBig().Uint64(),
+			ResourceType: res.ResourceType,
+			Count:        &count,
+		})
+	}
+
+	if tac.Spec.TiDB != nil {
+		strategy.Rules = append(strategy.Rules, autoRulesToStrategyRule("tidb", tac.Spec.TiDB.Rules))
+	}
+
+	if tac.Spec.TiKV != nil {
+		strategy.Rules = append(strategy.Rules, autoRulesToStrategyRule("tikv", tac.Spec.TiKV.Rules))
+	}
+	return strategy
+}
+
+func autoRulesToStrategyRule(component string, rules map[corev1.ResourceName]v1alpha1.AutoRule) *pdapi.Rule {
+	result := &pdapi.Rule{
+		Component: component,
+	}
+	for res, rule := range rules {
+		switch res {
+		case corev1.ResourceCPU:
+			result.CPURule = &pdapi.CPURule{
+				MaxThreshold:  rule.MaxThreshold,
+				MinThreshold:  *rule.MinThreshold,
```

Comment:
This should be defaulted so we will not dereference nil pointer

Reply:
How about adding a comment to explain why the min and max have different types?